watertap/costing/watertap_costing_package.py

```diff
@@ -1317,3 +1357,37 @@ def cost_by_flow_volume(blk, flow_cost, flow_to_cost):
     blk.capital_cost_constraint = pyo.Constraint(
         expr=blk.capital_cost == blk.flow_cost * flow_to_cost
     )
+
+
+def cost_uv_aop_bundle(blk, reactor_cost, lamp_cost, factor_uv_lamp_replacement):
+    """
+    Generic function for costing a UV system.
+
+    Args:
+        reactor_cost - The cost of UV reactor in [currency]/[volume]
+        lamp_cost - The costs of the lamps, sleeves, ballasts and sensors in [currency]/[kW]
+    """
+    make_capital_cost_var(blk)
+    make_fixed_operating_cost_var(blk)
+    blk.reactor_cost = pyo.Expression(expr=reactor_cost)
+    blk.lamp_cost = pyo.Expression(expr=lamp_cost)
+    blk.factor_uv_lamp_replacement = pyo.Expression(expr=factor_uv_lamp_replacement)
+
+    flow_in = pyo.units.convert(
+        blk.unit_model.control_volume.properties_in[0].flow_vol,
+        to_units=pyo.units.m**3 / pyo.units.hr,
+    )
+
+    electricity_demand = pyo.units.convert(
+        blk.unit_model.electricity_demand[0], to_units=pyo.units.kW
+    )
+
+    blk.capital_cost_constraint = pyo.Constraint(
+        expr=blk.capital_cost
+        == blk.reactor_cost * flow_in * blk.unit_model.num_of_reactors
+        + blk.lamp_cost * electricity_demand

```

Comment:
I'm a little unsure about this. This seems like the control volume for your UV model only accounts for 1 reactor, so you multiply by number of (parallel) reactors to accommodate a system with higher capacity.... but then the lamp_cost is multiplied by power demand, which I would expect to increase as well with number of reactors. Moreover, couldn't our control volume represent the whole system of reactors without having to ever specify number of reactors?

Reply:
This is a pretty good point! Actually different uv reactors have different numbers of lamps and other equipments, but as Tim suggested, I'd better not add so many variablles such as number of lamps, etc. I think for a simple solution, we could delete this `num_of_reactors` variable and use our control volume to represent the whole uv system as you suggested.

Reply:
OK, then it seems this is the only further change needed before merging, considering that you already addressed my other comments.

Reply:
_Unless_ the number of reactors accounts for redundant reactors that aren't used all at the same time. Then I could understand why only 1 reactor is considered for energy demand, while 3 (in your test examples) are considered for capital cost, which could represent 1 operational reactor and 2 redundant.

Reply:
@adam-a-a in the given example, there are 3 parallel reactors and 1 is redundant, there are 15 average number of lamps working over the simulation. Overall, the energy demand depends on the working lamps but the capital cost includes the redundant reactor, lamps and other equipments.